liquid-staking/src/setup/vote.rs

```diff
@@ -0,0 +1,52 @@
+multiversx_sc::imports!();
+
+use crate::{
+    basics::errors::{ERROR_ALREADY_VOTED, ERROR_INVALID_CALLER, ERROR_VOTE_SC_NOT_SET},
+    setup::governance,
+};
+#[multiversx_sc::module]
+pub trait VoteModule:
+    governance::GovernanceModule
+    + multiversx_sc_modules::default_issue_callbacks::DefaultIssueCallbacksModule
+{
+    #[endpoint(delegateVote)]
+    fn delegate_vote(
+        &self,
+        proposal: u32,
+        vote_type: ManagedBuffer,
+        delegate_to: ManagedAddress,
+        voting_power: BigUint,
+    ) {
+        let caller = self.blockchain().get_caller();
+
+        require!(!self.vote_contract().is_empty(), ERROR_VOTE_SC_NOT_SET);
```

Comment:
You could take these lines in a separate function (check_caller_is_vote_contract). Also, for clarity, instantiate a vote_contract_mapper variable.